torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -240,14 +252,26 @@ bool ProcessGroupNCCL::WorkNCCL::isSuccess() const {
 }
 
 void ProcessGroupNCCL::WorkNCCL::checkAndSetException() {
-  if (exception()) {
+  auto existingException = exception();
+  if (existingException) {
+    auto existingExceptionMsg = getExceptionMsgFromExceptionPtr(existingException);
+    LOG(INFO) << "[Rank " << rank_ << "]"
+              << " already had existing exception: " << existingExceptionMsg
+              << " when checking for NCCL communicator errors.";
     // We already have an exception.
```

Comment:
Do we need to LOG here again for the same rank? We would already have a LOG when we set the exception for the first time below right?

Reply:
That's a good point, we've already logged this one and there's no reason to log it again.